src/Games/NexusMods.Games.AdvancedInstaller.UI/Content/Left/ModContent/TreeEntry/IModContentTreeEntryViewModel.cs

```diff
@@ -0,0 +1,149 @@
+    /// </summary>
+    ReactiveCommand<Unit, Unit> CancelSelectCommand { get; }
+
+    /// <summary>
+    /// Command invoked when the user clicks the (X location) or (X Included) button.
+    /// Removes the mapping for this entry and child entries mapped with this entry.
+    /// </summary>
+    ReactiveCommand<Unit, Unit> RemoveMappingCommand { get; }
+
+    /// <summary>
+    /// Sets the mapping information for this entry and changes the state.
+    /// </summary>
+    /// <param name="entry">Matching Preview tree entry</param>
+    /// <param name="mappingFolderName">The name of the parent destination folder to display in the unlink button.</param>
+    /// <param name="isExplicit">Whether this is an explicit mapping, or a child being mapped through a parent.</param>
+    public void SetFileMapping(IPreviewTreeEntryViewModel entry, string mappingFolderName, bool isExplicit);
+
+    /// <summary>
+    /// Removes mapping information from this entry and changes the state.
+    /// !Doesn't remove mapping information from child entries.
+    /// </summary>
+    public void RemoveMapping();
+
+    /// <summary>
+    /// The invalid relative path used for the root entry.
+    /// Necessary for DynamicData TransformToTree, we need a RelativePath that is guaranteed not to represent another node.
+    /// </summary>
+    public static readonly RelativePath RootParentRelativePath = new("*rootParent*");
+}
+
+/// <summary>
+///     Represents the current status of the <see cref="NexusMods.Games.AdvancedInstaller.UI.ModContent.IModContentTreeEntryViewModel" />.
+/// </summary>
+public enum ModContentTreeEntryStatus : byte
+{
+    /// <summary>
+    ///     Item is not selected, and available for selection.
+    /// </summary>
+    Default,
+
+    /// <summary>
+    ///     The item target is currently being selected/mapped.
+    ///     This is used by the item which is currently being mapped into an install location.
+    /// </summary>
+    Selecting,
+
+    /// <summary>
+    ///     A parent of this item (folder) is currently being selected/mapped.
+    /// </summary>
+    /// <remarks>
+    ///     When this state is active, the UI shows 'include' for files, and 'include folder' for folders.
+    /// </remarks>
+    SelectingViaParent,
+
+    /// <summary>
+    ///     Item is included, with explicit target location.
+    /// </summary>
+    /// <remarks>
+    ///     When this state is active, the UI usually shows the name of the linked folder in the associated button.
+    /// </remarks>
+    IncludedExplicit,
+
+    /// <summary>
+    ///     Item id included, because a parent (folder) of the item is included.
+    ///     When the parent is unlinked, this node is also unlinked.
+    /// </summary>
+    /// <remarks>
+    ///     This is used to indicate a parent of this item which which is a directory has status
+    ///     <see cref="IncludedExplicit" />.
+    /// </remarks>
+    IncludedViaParent
+}
```

Comment:
This should go into its own file. Making the underlying type a `byte` is usually good practice if the type is going to be serialized. I'm not sure if that's ever going to be the case, but it never hurts to explicitly set a value for each member.